crates/move-stackless-bytecode/src/function_target_pipeline.rs

```diff
@@ -409,15 +414,133 @@ impl FunctionTargetsHolder {
             }
         }
 
-        func_env.get_name_str().strip_suffix("_inv").map(|name| {
-            if let Some(struct_env) = func_env
-                .module_env
-                .find_struct(func_env.symbol_pool().make(name))
-            {
-                self.datatype_invs
-                    .insert(struct_env.get_qualified_id(), func_env.get_qualified_id());
+        if let Some(spec_attr) = func_env
+            .get_toplevel_attributes()
+            .get_(&Verification(VerificationAttribute::SpecOnly))
+        {
+            let inner_attrs = match &spec_attr.value {
+                Attribute_::Parameterized(_, inner_attrs) => inner_attrs,
+                _ => &UniqueMap::new(),
+            };
+
+            let is_inv_target: bool =
+                inner_attrs.contains_key_(&AttributeName_::Unknown(Symbol::from("inv_target")));
+
+            if is_inv_target {
+                let function_spec = inner_attrs
+                    .get_(&AttributeName_::Unknown(Symbol::from("inv_target")))
+                    .unwrap();
+                if let Attribute_::Assigned(_, boxed_value) = &function_spec.value {
+                    if let AttributeValue_::ModuleAccess(spanned) = &boxed_value.value {
+                        if let ModuleAccess_::ModuleAccess(module_ident, struct_spanned) =
+                            &spanned.value
+                        {
+                            let address = module_ident.value.address;
+                            let module = &module_ident.value.module;
+                            let struct_name = struct_spanned.value.as_str();
+
+                            let addr_bytes = address.into_addr_bytes();
+                            let module_name = ModuleName::from_address_bytes_and_name(
+                                addr_bytes,
+                                func_env.symbol_pool().make(&module.to_string()),
+                            );
+
+                            if let Some(module_env) =
+                                func_env.module_env.env.find_module(&module_name)
+                            {
```

Comment:
let's extract this in a helper function `parseModuleAccess`, it's the same code as for function